framework-test/src/main/java/org/checkerframework/framework/test/TestConfigurationBuilder.java

```diff
@@ -67,6 +67,8 @@ public static TestConfigurationBuilder getDefaultConfigurationBuilder(
             configBuilder.addOption("-Xbootclasspath/p:" + jdkJarPath);
         }
 
+        configBuilder.addOption("-source", "8").addOption("-target", "8");
```

Comment:
Thanks for tracking this down!
Can you model this similarly to https://github.com/eisop/checker-framework/blob/master/framework/src/main/java/org/checkerframework/framework/util/CheckerMain.java#L435
and add a check whether we're running on a 8 JVM?
Include the `-Xbootclasspath` part above.

Can you search the whole codebase for `-Xbootclasspath` and 1) make sure there is a source/target 8 nearby and 2) that it is guarded by a check for a 8 JVM.

Reply:
So add a similar check in TypecheckExecutor.compile. We can later try to untangle the duplication between the two.